pkg/deploy/graph/analysis/dc.go

```diff
@@ -48,6 +50,13 @@ dc:
 					continue dc
 				}
 
+				if bcNode, ok := buildedges.BuildPointsToTag(g, istNode); ok && !istNode.Found() {
+					// Avoid warning for the dc image trigger in case there is a build in flight.
+					if latestBuild := buildedges.GetLatestBuild(g, bcNode); latestBuild != nil && !buildutil.IsBuildComplete(latestBuild.Build) {
+						continue dc
```

Comment:
We're seven levels of indentation deep here and using a named `continue` target - do you think we can break this up into smaller pieces?